Assets/com.fluid.behavior-tree/Runtime/TaskParents/Composites/Parallel.cs

```diff
@@ -1,52 +1,25 @@
-﻿using System.Collections.Generic;
 using CleverCrow.Fluid.BTs.Tasks;
 
 namespace CleverCrow.Fluid.BTs.TaskParents.Composites {
     public class Parallel : CompositeBase {
-        private readonly Dictionary<ITask, TaskStatus> _childStatus = new Dictionary<ITask, TaskStatus>();
-
         public override string IconPath { get; } = $"{PACKAGE_ROOT}/CompareArrows.png";
 
         protected override TaskStatus OnUpdate () {
-            var successCount = 0;
-            var failureCount = 0;
-
-            foreach (var child in Children) {
-                TaskStatus prevStatus;
-                if (_childStatus.TryGetValue(child, out prevStatus) && prevStatus == TaskStatus.Success) {
-                    successCount++;
-                    continue;
-                }
-
-                var status = child.Update();
-                _childStatus[child] = status;
-
-                switch (status) {
-                    case TaskStatus.Failure:
-                        failureCount++;
-                        break;
-                    case TaskStatus.Success:
-                        successCount++;
-                        break;
-                }
-            }
-
-            if (successCount == Children.Count) {
-                End();
-                return TaskStatus.Success;
-            }
-
-            if (failureCount > 0) {
+            TaskStatus status = TaskStatus.Success;
+            foreach(var child in Children)
+                status |= child.Update();
+            if (TaskStatus.Failure == (status & TaskStatus.Failure)) {
                 End();
                 return TaskStatus.Failure;
             }
-
-            return TaskStatus.Continue;
+            else if(TaskStatus.Continue == (status & TaskStatus.Continue)) {
```

Comment:
Should just be `if (status == TaskStatus.Continue) {`. Also, this is the most likely scenario so let's make it the first if check.